Using the size of the tabs as minimum size.

George-Devel opened the pull request:
See Netbeans issue #3710

George-Devel commented:
Hi, my code change may look weird, but this component is responsible for drawing the tab-bar in the editor area. This is only the part of the "North" component of a border layout. The minimum size of the actual content is the "Center" component and is considered by the panel.

You can check, that this works correctly by trying the Platform Application attached to the bug.
It contains only one relevant file: an editor window, registered with annotations, with a minimum size set and an overridden paint method to show the minimum size. If you like I can this file separately.

DevCharly commented on platform/o.n.swing.tabcontrol/src/org/netbeans/swing/tabcontrol/plaf/BasicScrollingTabDisplayerUI.java:
```diff
@@ -468,20 +468,7 @@ protected Rectangle getControlButtonsRectangle( Container parent ) {
-        minSize.width = Math.max(minSize.width, 100);
-        minSize.height = Math.max(minSize.height, 10);
-        return minSize;
+        return new Dimension( 100,  displayer.getSize().height);
```
It would be better to use `displayer.getPreferredSize().height` here.
`displayer.getSize()` returns the current size, which is initially `0,0` before first layout pass.

George-Devel replied:
Ok, I will change this. I'll also have look at `AbstractViewTabDisplayerUI `. I see that this is used if "Multi-row tabs" is selected in the Appearance options . It seems that `AbstractViewTabDisplayerUI.getMinimumSize` returns height of a single tab even then multiple rows are displayed.

DevCharly reviewed (changes requested):
I agree that `BasicScrollingTabDisplayerUI.getMinimumSize()` should return only the minimum size of the tabs.
Currently it returns the the minimum size of the tabs plus the minimum size of the tab content.
`getPreferredSize()` in `BasicScrollingTabDisplayerUI` subclasses also return only preferred size of the tabs.

For some unknown reason, this was added in 2008 in commit: https://github.com/emilianbold/netbeans-releases/commit/82a48e93252068b86905bcbfac1e7e508e6aa171#diff-2f04bb43144858669541be39da8de41c29eef37380e8a2b3ce26dce479c92534
This is the related issue: https://bz.apache.org/netbeans/show_bug.cgi?id=136636

Don't understand the need for this `BasicScrollingTabDisplayerUI.getMinimumSize()` implementation. Maybe some part of the windowing system worked different in 2008 that it does today...

There is a similar `getMinimumSize()` method in `AbstractViewTabDisplayerUI`, which was added in the same commit:
https://github.com/emilianbold/netbeans-releases/commit/82a48e93252068b86905bcbfac1e7e508e6aa171#diff-c2ceef123f96cb6d9bbb42f06afa4cdc636d72746b85139890a4774e6caddac2

Maybe this also needs to be changed? But maybe only if `isUseStretchingTabs()` returns `true`.

George-Devel commented:
@DevCharly - I changed the line as requested. I also replaced 'getMinimumSize()' in `AbstractViewTabDisplayerUI` with the same code. I checked the 'BasicScrollingTabDisplayerUI' in the Editor area, and  `AbstractViewTabDisplayerUI`  in the output area ,including the variant with "winsys.stretching_view_tabs" set to true.

My assumption that the `AbstractViewTabDisplayerUI` is used if "Multi-row tabs" was wrong.  This another independent issue.

eirikbakke reviewed (approved):
Thanks for reviewing, @DevCharly! The change sound fine with me... I'll test this in my IDE and NetBeans Platform application the next time I do a build, but no need to wait for that.

lkishalmi commented:
Well, it is approved. Let's merge it then!
Thank you @George-Devel for your contribution!
